src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -482,44 +482,62 @@ public static bool TryValidateBlock(BinaryReader reader, ZipArchiveEntry entry)
             }
             else
             {
-                if (reader.BaseStream.Length < reader.BaseStream.Position + extraFieldLength + entry.CompressedLength + 4)
+                if (reader.BaseStream.Length < reader.BaseStream.Position + extraFieldLength + entry.CompressedLength + 12)
                 {
                     return false;
                 }
 
                 reader.BaseStream.Seek(extraFieldLength + entry.CompressedLength, SeekOrigin.Current); // seek to end of compressed file from which Data descriptor starts
-                uint dataDescriptorSignature = reader.ReadUInt32();
-                bool wasDataDescriptorSignatureRead = false;
-                int seekSize = 0;
-                if (dataDescriptorSignature == DataDescriptorSignature)
+                uint[] buffer = new uint[6];
+                // let's try to match the smallest possible structure (3 x 4) 32 bit without signature
```

Comment:
Does it make sense to skip these if the size from central dir is greater than Unit32.MaxValue?

Reply:
By `skip these` you mean skip validation?

Reply:
I mean skip reading as 32-bits if you know the sizes cannot be represented as 32-bits.  You’d still validate but only consider 64-but format.

Reply:
Ah I see, @eerhardt actually come up with a similar solution https://github.com/dotnet/runtime/compare/master...eerhardt:FixZipIssues.
So now the question is if we want to go with @eerhardt solution + AppContext switch or WPF solution + AppContext switch. WPF solution worked with the 17GB zip locally, but didn't added a test for that, too big, the zip entry alone is 3GB of compressed size.

For me to avoid for any other strange edge case we might better use WPF solution which has more forgiving logic than testing exact matching case we have now. If we remove this condition on row 516 even `@MikeCodesDotNET`'s buggy zip will be passed and we might not need to add AppContext switch to disable validation